drivers/spi/CMakeLists.txt

```diff
@@ -49,7 +49,10 @@ zephyr_library_sources_ifdef(CONFIG_SPI_AMBIQ		spi_ambiq.c)
 zephyr_library_sources_ifdef(CONFIG_SPI_AMBIQ_BLEIF	spi_ambiq_bleif.c)
 zephyr_library_sources_ifdef(CONFIG_SPI_RPI_PICO_PIO	spi_rpi_pico_pio.c)
 zephyr_library_sources_ifdef(CONFIG_SPI_MCHP_MSS	spi_mchp_mss.c)
-zephyr_library_sources_ifdef(CONFIG_SPI_RTIO spi_rtio.c)
+zephyr_library_sources_ifdef(CONFIG_SPI_RTIO
+	spi_rtio.c
+	spi_rtio_default.c
```

Comment:
Hi @tbursztyka thanks for your review!

May we reconsider this point? Let me explain my rationale:

`spi_rtio.c` and `spi_rtio_default.c` should have completely different purposes:
- spi_rtio.c/spi_rtio.h would be to have common functionality that drivers implementing RTIO _natively_ would use (this goes in-line with what i2c_rtio does, and there's an open PR for doing this on SPI (#77136).
- OTOH,  spi_rtio_default is the "emulated" implementation if you don't have a native one.
- These two are mutually exclusive: either you have a native implementation (so you don't need the default), or you don't (so you wouldn't need spi_rtio.c).

As such, I think lumping them together in the same file will be counter-intuitive to users.

LMK what you think!

Reply:
> * These two are mutually exclusive: either you have a native implementation (so you don't need the default), or you don't (so you wouldn't need spi_rtio.c).

But both would be compiled anyway, and user will anyway need to include spi_rtio.h also.

If we take things as: whatever belongs to rtio (whether it's native or not) should go into spi_rtio.c/spi_rtio.h, it is as intuitive as it can get.

Also, there is an issue with "default", semantically it is not self-describing. Better name the function : 
spi_rtio_iodev_emulated_submit() 

Add your copyright along with the current one in spi_rtio.c also. (and in spi_rtio.h obviously)

Reply:
Fair enough. I've merged both into `spi_rtio.c`